tests/test_web_protocol.py

```diff
@@ -1,805 +0,0 @@
-# Tests for aiohttp/server.py
```

Comment:
Why are you removing these tests completely?

Reply:
Because they are mock-based totally.
These tests test their own mocks, nothing else.
I wrote these tests striving to get 100% formal coverage but I see later that tests don't help find any error.
The code is covered by functional tests more or less, which is a good enough compromise.

BTW, the same situation exists for asyncio test suite: a part of tests a pure mock-based one. They were useful at the very beginning but now when a mocked test fails by a change of internal logic or refactoring -- we just kick off this subtle useless test.

Reply:
It looks like the coverage is reduced. I guess my point is that the functional tests should make sure to cover the same lines this suite used to cover.

Reply:
Fair point!
The coverage was reduced because handling the base `Exception` thrown from HttpParser was not covered by tests.
Ironically, the code that returns 500 on the internal parser error was invalid.
The correct behavior in this case -- do nothing but catch the exception by asyncio code, report about an unexpected exception, and close the transport.
The wrong code is dropped, thank you.
Previously we had a mock-based test that emulated the error-ness behavior without careful thinking.